lib/Service/ColumnService.php

```diff
@@ -325,4 +331,42 @@ public function delete(int $id, bool $skipRowCleanup = false, ?string $userId =
 			throw new InternalError($e->getMessage());
 		}
 	}
+
+	/**
+	 * @param array $titles example ['Test column 1', 'And so on', '3rd column title']
+	 * @throws PermissionError
+	 * @throws InternalError
+	 *
+	 * @return array with column object or null for given columns
+	 */
+	public function findColumnsByTitleForTableAsArray(int $tableId, array $titles, ?string $userId, bool $createUnknownColumns, int &$countCreatedColumns): array {
+		$result = [];
+
+		if($userId === null) {
+			$userId = $this->userId;
+		}
+		$allColumns = $this->findAllByTable($tableId, $userId);
+		$i = -1;
+		foreach ($titles as $title) {
+			$i++;
+			foreach ($allColumns as $column) {
+				if($column->getTitle() === $title) {
+					$result[$i] = $column;
+					continue 2;
+				}
+				$result[$i] = '';
+			}
+			// if there are no columns at all
+			if(!isset($result[$i])) {
+				$result[$i] = '';
+			}
+			// if column was not found
+			if($result[$i] === '' && $createUnknownColumns) {
+				$description = $this->l->t('This column was automatically created by the import service.');
+				$result[$i] = $this->create($userId, $tableId, 'text', 'line', $title, false, $description, null, null, null, null, null, null, null, null, null, null, null, null, null);
```

Comment:
Naming: with the method called `find…` i do not expect that anything will be modified here. Would split the logic, or change the naming,